syft/workers/websocket_client.py

```diff
@@ -71,13 +71,31 @@ def search(self, *query):
         response = self._recv_msg(serialized_message)
         return sy.serde.deserialize(response)
 
+    def get_ptr(self, obj_id):
+        message = messaging.PlanCommandMessage((obj_id,), "get_ptr")
+        serialized_message = sy.serde.serialize(message)
+        response = self._recv_msg(serialized_message)
+        return sy.serde.deserialize(response)
+
     def fetch_plan(self, plan_id):
-        # Prepare a message requesting the websocket server to search among its objects
-        message = messaging.FetchPlanMessage(plan_id)
+        message = messaging.PlanCommandMessage((plan_id,), "fetch_plan")
         serialized_message = sy.serde.serialize(message)
         # Send the message and return the deserialized response.
         response = self._recv_msg(serialized_message)
-        return sy.serde.deserialize(response)
+        plan = sy.serde.deserialize(response)
+
+        if plan.state_ids:
+            state_ids = []
+            for state_id in plan.state_ids:
+                ptr = self.get_ptr(state_id)
+                ptr.owner = sy.hook.local_worker
+                sy.hook.local_worker._objects[ptr.id] = ptr
```

Comment:
That's not OK haha

Ok so I see what you need, instead of creating def get_ptr, you should use `register_response` of `hook_args` (extract of the doctring: "When a remote worker execute a command sent by someone else, the response is inspected: all tensors are stored by this worker and a Pointer tensor is made for each of them.").
this is what is used in baseworker.py::execute_command, it's very practical and you have nothing to care about.

Reply:
I'm not sure how to use `register_response`, can you send a snippet with pseudo-code of how that would look like?

From the documentation + example in base I understand that `base.get_ptr` should use register_response like this:

```
    def get_ptr(self, obj_id: Union[str, int]):
        """Returns a pointer to an object from the registry.

        Args:
            obj_id: A string or integer id of an object to look up.
        """
        obj = self.get_obj(obj_id)
        if hasattr(obj, "allowed_to_get") and not obj.allowed_to_get():
            raise GetNotPermittedError()
        else:
            # response = obj.create_pointer(garbage_collect_data=False, owner=sy.local_worker).wrap()
            ptr = sy.frameworks.torch.hook_args.register_response(
                "get_ptr", obj, list(tuple([sy.ID_PROVIDER.pop()])), self
            )
            return ptr
```

And I don't need `websocket_client.get_ptr` anymore, so `websocket.fetch_plan` will look like this, is that right?

```
    def fetch_plan(self, plan_id):
        message = messaging.PlanCommandMessage((plan_id,), "fetch_plan")
        serialized_message = sy.serde.serialize(message)
        # Send the message and return the deserialized response.
        response = self._recv_msg(serialized_message)
        plan = sy.serde.deserialize(response)

        if plan.state_ids:
            state_ids = []
            for state_id in plan.state_ids:
                ptr = self.get_ptr(state_id)
                sy.hook.local_worker._objects[ptr.id] = ptr
                state_ids.append(ptr.id)
            plan.replace_ids(plan.state_ids, state_ids)
            plan.state_ids = state_ids

        plan.replace_worker_ids(self.id, sy.hook.local_worker.id)
        return plan
```

Reply:
Hey Theo, can you check if this is better? https://github.com/OpenMined/Grid/pull/288